packages/webapp/src/elections/components/ongoing-election-components/ongoing-round-segment.tsx

```diff
@@ -200,14 +203,7 @@ export const OngoingRoundSegment = ({
         setIsSubmittingVote(false);
     };
 
-    // TODO: this needs more attention once we have sufficient test data to test it
-    if (!participants?.length) {
-        return (
-            <div>
-                You nor a delegate of yours is participating in this round.
-            </div>
-        );
-    }
+    if (!ongoingElectionData?.isMemberStillParticipating) return null;
```

Comment:
This is the only place we're using `ongoingElectionData` in this component (we will use it a lot more in this component soon, but that's for later!) And we're doing >200 lines worth of queries and calculations above it only to return `null` depending on the answer to a question, which we already know. 

I recommend removing (for now) `ongoingElectionData` from the props of this component and doing this check instead inside the `CurrentRound` component that appears on line `167` of `ongoing-election.tsx`. We can short circuit a lot of this that way.

Reply:
There's no getting around doing those 200 lines of code. They'll be gotten by 5 other components. And it's all cached. So extremely small cost, *and* given another of your comments above, I now use 2 fields from the structure, so I'd greatly prefer to leave in the nicer, more readable/maintainable structure

Reply:
I suspect we're misunderstanding one another's comments. My point is more: Why invoke this component at all if `!isMemberStillParticipating`?